### PR TITLE
Add Filter<T>, fix filterValueFormatters not being used in PostFilter<T>

### LINKED ISSUE #45: Create Filter<T> as a sibling to PostFilter<T>
In [src/Filter.Nest/SearchDescriptorExtensions.cs](https://github.com/ritterim/filter/blob/master/src/Filter.Nest/SearchDescriptorExtensions.cs) create a `Filter<T>` as a sibling to `PostFilter<T>`.

- `Filter<T>` can compute relevancy score for the results that will be returned (rather than all results returned prior to using `PostFilter<T>`).
- This may be useful with aggregation concerns as well.

### PR DESCRIPTION
**Pull request by kendaleiv**
Resolves #45.

@khalidabuhakmeh and I worked on this. We'd like an objective third party review.

### CODE REVIEW

File: src/Filter.Nest/QueryContainerDescriptorExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+{
+    public static class QueryContainerDescriptorExtensions
+    {
+        public static QueryContainer Filter<T>(
+            this QueryContainerDescriptor<T> value,
+            object filter,
+            IDictionary<Type, Func<object, string>> filterValueFormatters = null)
+            where T : class
+        {
+            if (filter == null)
```
Comment (billbogaiv) [inline src/Filter.Nest/QueryContainerDescriptorExtensions.cs]: This majority of the method body is the same as [SearchDescriptorExtensions](https://github.com/ritterim/filter/blob/497cf748dd52d4d00631d550ea1ba35f56ee8585/src/Filter.Nest/SearchDescriptorExtensions.cs#L24-L103). If we expect both types of `Filter` to have the same behavior, why not just extract most of the body, give it a return-type of `List<Func<QueryContainerDescriptor<T>, QueryContainer>>`, and use the return-value [on this line](https://github.com/ritterim/filter/pull/46/files#diff-3b4683a3f2a756be47c20cca315a352fR90)?

Comment (khalidabuhakmeh) [inline src/Filter.Nest/QueryContainerDescriptorExtensions.cs]: This has been addressed. The filter logic has been moved to an internal class called `FilterLogic`.

Comment (billbogaiv) [inline src/Filter.Nest/QueryContainerDescriptorExtensions.cs]: Nice.

Kylo Ren understands the meaning of teamwork

**Comment by billbogaiv** [comment]
@cathy-king: if everything looks good to you, let me know. I'm good with merging this PR.

**Comment by khalidabuhakmeh** [comment]
This PR be like

**Pull request merged**